vsphere/resource_vsphere_virtual_disk.go

```diff
@@ -465,3 +482,41 @@ func searchForDirectory(client *govmomi.Client, datacenter string, datastore str
 
 	return nil
 }
+
+func resourceVSphereVirtualDiskImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*VSphereClient).vimClient
+	p := d.Id()
+	if !strings.HasPrefix(p, "/") {
+		return nil, errors.New("ID must start with a trailing slash")
+	}
+
+	// The path in the ID has the form `/<datacenter>/[<datastore>] path/to/vmdk`.
+	// Note that the values we care about in addrParts start at the first element as
+	// the zero-th element will be empty (that is, everything before the / prefix).
+	addrParts := strings.SplitN(p, "/", 3)
```

Comment:
I think this should be increased to four as the expected path should be `/<dc-name>/datastore/<datastore-name>/path/to/datastore` to match the govc/vSphere convention.

Reply:
Hi Bill,
Thanks for the suggestion. Perfectly happy to change to your suggested format, but just wanted to explain why I'd gone for the format that I had, just in case it changed your mind!
When I look in the vSphere web UI at the file associated with the virtual disk, the path that it provides is of the form I was suggesting (that is, `[<datastore>] path/to/vmdk`) so seemed like a sensible choice for what to use for the import. I did start with something closer to what you have suggested based on the format for other import commands, but changed to what is implemented currently after seeing the format in the UI. 
If I haven't convinced you, just let me know, and I'm happy to tweak the parameter parsing code accordingly.
Thanks,
Richard.

Reply:
Hi Bill,
Looking at this a bit more today, I think the interface into govc (via govmomi's object.VirtualDiskManager) does use the format I am proposing, so I guess it becomes a choice between matching the govc/vSphere conventions or being more inline with the format for importing other resource types (or maybe those two things are actually the same, and it is just the format of an inventory path for a virtual disk (that is, a datastore path) is very different to the path for other resource types).
Would appreciate your thoughts here, following my comments. Are you happy for me to stick with `/<dc-name>/[<datastore-name>] /path/to/vmdk`, or would you still prefer me to change to `/dc-name>/datastore/<datastore-name>/path/to/vmdk` and convert from the latter into the former in the provider code?
Thanks,
Richard.

Reply:
I guess I was mistaken on the govc side. Since thats the case, lets stick with the format you used initially. Thanks for the follow up!